### ISSUE TITLE
Fix documentation of addons

### ISSUE THREAD

**Issue by brodybits** [jxcore/jxcore#549]
The existing documentation of addons came from Node (0.10) and has not been updated to reflect the API of this project. I am especially interested in the "Callbacks", "Factory of wrapped objects", and "Passing wrapped objects around" sections. While it would be possible to find examples looking through the source code and https://github.com/Level/leveldown-mobile (some of which are using a customized version of Node NAN), having these properly documented in one place would be much better.

A couple other points:
- `doc/native` seems to be a mixture of multiple layers: embedded JXCore API and general native JX object API, where the general native JX object API would apply to addons and internal code as well. I wonder if we could find a better way to organize this?
- It would be _really_ nice to have a version of the Node NAN that allows people to use the same API to build native modules for both Node.js and this project. I noticed that https://github.com/Level/leveldown-mobile/blob/master/src/nan.h does this to a very limited extent, would be nice if it would support the full API. I tried this once and thinking it may be just as well to start again from the NAN 2.0.x source.

**Comment by obastemur** [comment]
> doc/native seems to be a mixture of multiple layers: embedded JXCore API and general native JX object API, where the general native JX object API would apply to addons and internal code as well. I wonder if we could find a better way to organize this?

Maybe we can put them under separate folders. Addon and Embedding.

>  It would be really nice to have a version of the Node NAN that allows people to use the same API to build native modules for both Node.js and this project.

Unfortunately NAN falls short on data types and type conversions. There is an ongoing discussion for public Node API.

**Comment by brodybits** [comment]
I have another question: how does JXCore ensure that it uses libuv in a thread-safe manner?

**Comment by obastemur** [comment]
Each individual JXcore instance has its own loop variable  +  some internal changes to libUV

**Comment by brodybits** [comment]
Can you give me some pointers where to look for the internal changes to libUV?

**Comment by obastemur** [comment]
Your best friend is a `diff` tool though. Compare to one in 0.10.x.

**Comment by brodybits** [comment]
Right. What do you think about getting this into a real libuv fork, or _maybe_ even libuv itself, so it can help some others?

**Comment by brodybits** [comment]
Also, forgot to mention that I would be happy to help with these items, whenever I get a chance.

**Comment by brodybits** [comment]
Some additional notes:
- #617 was raised for JX_Persistent() documentation
- I did not realize that JX has a C API until reading https://github.com/jxcore/jxcore/issues/617#issuecomment-149220930
- While developing the https://github.com/brodybits/node-uvhttp addon with a wrapped object factory as documented in https://nodejs.org/api/addons.html (using NAN), I had to make a per-thread V8 constructor object to work with multi-threading. I will raise a PR to document this when I get a chance.
